### PR TITLE
Ported Skyrat's languages menu.

### PR DESCRIPTION
**Pull request by lizardqueenlexi**
## About The Pull Request

I've directly ported the language menu from Skyrat to our codebase. This adds a new tab to the character prefs window, "Languages", that provides a list of languages and allows you to select up to three of them.

image

Your languages default to those from your species, but you are not required to keep any of the defaults. You can even decide not to speak Galactic Common if you really want to. (For this reason, I've _also_ disabled the Foreigner quirk, as it would be redundant). Finally, there's a new positive quirk, "Linguist", that lets you learn a fourth language.

Note that a number of languages have been flagged as "secret", disallowing them from being selected on this menu. These languages are: Aphasia, Codespeak, Drone, Encoded Audio Language, Mushroom, Narsian, Piratespeak, Shadowtongue, and Xenocommon. If anyone has any other opinions on what should go on this list, let me know.

Please feel free to provide other suggestions. This feature has been largely ported directly, but that doesn't mean we can't make our own changes to it.

## Why It's Good For The Game

The current language system is too rigid and not always good for RP - having languages _strictly_ based on your own species is both limiting and extremely D&D-like, something we probably don't particularly want. With this menu, you can select your character's languages how you like, which might allow for some fun storytelling.

A potential downside is that you can no longer _rely_ on someone speaking a specific language due to their species, but that's not exactly a key feature of this game to begin with. This may also _slightly_ infringe on the Curator's ability to speak every language, but you know. They still get _every single one_, including the secret ones. I think it's fine.

## Changelog

:cl:
add: Added the Languages menu from Skyrat.
del: Removed the Foreigner quirk.
/:cl:

### CODE REVIEW

File: code/datums/quirks/neutral.dm
Comment (skylord-a52) [inline code/datums/quirks/neutral.dm:37]: Rather than removing a quirk wholesale, it might be better to add a "ENABLED/DISABLED" flag so that we can remove them non-intrusively. But that's a bit out of the scope of this PR and it's been waiting for approval for two weeks, so I'll leave it up to you whether or not you want to actually implement that here.

**Review by skylord-a52 — APPROVED** [review]

File: code/datums/quirks/neutral.dm
Comment (lizardqueenlexi) [reply code/datums/quirks/neutral.dm:37]: I'll note this down for a later implementation!

**Pull request merged**
